controllers/dnspolicy_controller_multi_cluster_test.go

```diff
@@ -141,41 +142,50 @@ var _ = Describe("DNSPolicy Multi Cluster", func() {
 				err := k8sClient.List(ctx, recordList, &client.ListOptions{Namespace: testNamespace})
 				g.Expect(err).NotTo(HaveOccurred())
 				g.Expect(recordList.Items).To(HaveLen(2))
-				g.Expect(recordList.Items).To(
-					ContainElements(
-						MatchFields(IgnoreExtras, Fields{
-							"ObjectMeta": HaveField("Name", recordName),
-							"Spec": MatchFields(IgnoreExtras, Fields{
-								"OwnerID":        Equal(&ownerID),
-								"ManagedZoneRef": HaveField("Name", "mz-example-com"),
-								"Endpoints": ConsistOf(
-									PointTo(MatchFields(IgnoreExtras, Fields{
-										"DNSName":       Equal(TestHostOne),
-										"Targets":       ContainElements(TestIPAddressOne, TestIPAddressTwo),
-										"RecordType":    Equal("A"),
-										"SetIdentifier": Equal(""),
-										"RecordTTL":     Equal(externaldns.TTL(60)),
-									})),
-								),
-							}),
-						}),
-						MatchFields(IgnoreExtras, Fields{
-							"ObjectMeta": HaveField("Name", wildcardRecordName),
-							"Spec": MatchFields(IgnoreExtras, Fields{
-								"OwnerID":        Equal(&ownerID),
-								"ManagedZoneRef": HaveField("Name", "mz-example-com"),
-								"Endpoints": ConsistOf(
-									PointTo(MatchFields(IgnoreExtras, Fields{
-										"DNSName":       Equal(TestHostWildcard),
-										"Targets":       ContainElements(TestIPAddressOne, TestIPAddressTwo),
-										"RecordType":    Equal("A"),
-										"SetIdentifier": Equal(""),
-										"RecordTTL":     Equal(externaldns.TTL(60)),
-									})),
-								),
-							}),
-						}),
-					))
+
+				dnsRecord := &kuadrantdnsv1alpha1.DNSRecord{}
+				err = k8sClient.Get(ctx, client.ObjectKey{Name: recordName, Namespace: testNamespace}, dnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+
+				wildcardDnsRecord := &kuadrantdnsv1alpha1.DNSRecord{}
+				err = k8sClient.Get(ctx, client.ObjectKey{Name: wildcardRecordName, Namespace: testNamespace}, wildcardDnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+
+				g.Expect(*dnsRecord).To(MatchFields(IgnoreExtras, Fields{
+					"ObjectMeta": HaveField("Name", recordName),
+					"Spec": MatchFields(IgnoreExtras, Fields{
+						"OwnerID":        Equal(&ownerID),
+						"ManagedZoneRef": HaveField("Name", "mz-example-com"),
+						"Endpoints": ConsistOf(
+							PointTo(MatchFields(IgnoreExtras, Fields{
+								"DNSName":       Equal(TestHostOne),
+								"Targets":       ContainElements(TestIPAddressOne, TestIPAddressTwo),
+								"RecordType":    Equal("A"),
+								"SetIdentifier": Equal(""),
+								"RecordTTL":     Equal(externaldns.TTL(60)),
+							})),
+						),
+					}),
+				}))
```

Comment:
```suggestion
				g.Expect(dnsRecord.Name).To(Equal(recordName))
				g.Expect(dnsRecord.Spec.OwnerID).To(Equal(&ownerID))
				g.Expect(dnsRecord.Spec.ManagedZoneRef.Name).To(Equal("mz-example-com"))
				g.Expect(dnsRecord.Spec.Endpoints).To(ConsistOf(
					PointTo(MatchFields(IgnoreExtras, Fields{
						"DNSName":       Equal(TestHostOne),
						"Targets":       ContainElements(TestIPAddressOne, TestIPAddressTwo),
						"RecordType":    Equal("A"),
						"SetIdentifier": Equal(""),
						"RecordTTL":     Equal(externaldns.TTL(60)),
					})),
				))
```

The main reason these should be re-factored is that when they fail having so much done in a single `Expect` produces a really hard to read error, in some cases so large it doesn't even print all of it. Not doing a single expect on the entire list(`recordList.Items`) is an improvement, but it can be improved again by just doing an expect on each part of the dnsrecord we care about. 

This re-factor should probably be done in it's own PR where there are no additional changes to the tests or code.

Reply:
Easier to read: 

```.go
g.Expect(dnsRecord.Name).To(Equal(recordName))
g.Expect(dnsRecord.Spec.OwnerID).To(Equal(&ownerID))
g.Expect(dnsRecord.Spec.ManagedZoneRef.Name).To(Equal("mz-example-com"))
g.Expect(dnsRecord.Spec.Endpoints).To(ConsistOf(
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":       Equal(TestHostOne),
		"Targets":       ContainElements(TestIPAddressOne, TestIPAddressTwo),
		"RecordType":    Equal("A"),
		"SetIdentifier": Equal(""),
		"RecordTTL":     Equal(externaldns.TTL(60)),
	})),
))
```

Reply:
#652 to track it